infra/core/compute/ubuntu-jumphost.bicep

```diff
@@ -0,0 +1,277 @@
+// USER-DEFINED TYPES
+// ========================================================================
+
+// From: infra/types/DiagnosticSettings.bicep
+@description('The diagnostic settings for a resource')
+type DiagnosticSettings = {
+  @description('The number of days to retain log data.')
+  logRetentionInDays: int
+
+  @description('The number of days to retain metric data.')
+  metricRetentionInDays: int
+
+  @description('If true, enable diagnostic logging.')
+  enableLogs: bool
+
+  @description('If true, enable metrics logging.')
+  enableMetrics: bool
+}
+
+// ========================================================================
+// PARAMETERS
+// ========================================================================
+
+@description('The diagnostic settings to use for logging and metrics.')
+param diagnosticSettings DiagnosticSettings
+
+@description('The name of the primary resource')
+param name string
+
+@minLength(3)
+@maxLength(15)
+@description('The name of the linux PC.  By default, this will be automatically constructed by the resource name.')
+param computerLinuxName string?
+
+@description('Username for the Virtual Machine.')
+param adminUsername string
+
+@description('Type of authentication to use on the Virtual Machine. SSH key is recommended.')
+@allowed([
+  'sshPublicKey'
+  'password'
+])
+param authenticationType string = 'password'
+
+@description('SSH Key or password for the Virtual Machine. SSH key is recommended.')
+@secure()
+param adminPasswordOrKey string
+
+@description('The Ubuntu version for the VM. This will pick a fully patched image of this given Ubuntu version.')
+@allowed([
+  'Ubuntu-1804'
+  'Ubuntu-2004'
+  'Ubuntu-2204'
+])
+param ubuntuOSVersion string = 'Ubuntu-2204'
+
+@description('Location for all resources.')
+param location string = resourceGroup().location
+
+@description('The size of the VM')
+param vmSize string = 'Standard_B2ms'
+
+@description('Security Type of the Virtual Machine.')
+@allowed([
+  'Standard'
+  'TrustedLaunch'
+])
+param securityType string = 'TrustedLaunch'
+
+@description('The tags to associate with this resource.')
+param tags object = {}
+
+/*
+** Dependencies
+*/
+
+@description('The ID of the Log Analytics workspace to use for diagnostics and logging.')
+param logAnalyticsWorkspaceId string = ''
+
+@description('The subnet ID to use for the resource.')
+param subnetId string
+
+/*
+** Settings
+*/
+
+// ========================================================================
+// VARIABLES
+// ========================================================================
+
+var validComputerName = replace(replace(name, '-', ''), '_', '')
+var computerName = !empty(computerLinuxName) ? computerLinuxName : length(validComputerName) > 15 ? substring(validComputerName, 0, 15) : validComputerName
+
+var configScriptRepoUrl = 'https://raw.githubusercontent.com/KSchlobohm/reliable-web-app-vm-postconfiguration/main'
```

Comment:
Looking at [these docs](https://learn.microsoft.com/en-us/azure/virtual-machines/extensions/custom-script-windows) it looks like the preferred way to do this would be to make the file available in blob storage and run it from there. I expect pulling from our GitHub repo is ok, though, and it's likely simpler.

Reply:
added